File: java/algo/src/main/java/algs/collection/Stack.java
package algs.collection;

public class Stack {
    private final long[] array;
    private final int size;
    private int top;

    public Stack(int size) {
        array = new long[size];
        this.size = size;
        top = -1;
    }

    public void push(long elm) {
        array[++top] = elm;
    }

    public long pop() {
        return array[top--];
    }

    public long peek() {
        return array[top];
    }

    public boolean isEmpty() {
        return (-1) == top;
    }

    public boolean isFull() {
        return (size - 1) == top;
    }
}

File: python/algo/stack.py
# 时间复杂度分析
# push O(1)
# pop  O(1)
# size O(1)
# peek O(1)
class Stack:
    def __init__(self):
        self.items = []
    def isEmpty(self):
        return self.items == []
    def push(self, item):
        self.items.append(item)
    def pop(self):
        return self.items.pop()
    def peek(self):
        return self.items[len(self.items)-1]
    def size(self):
        return len(self.items)

if '__main__' == __name__:
    stack = Stack()
    stack.push('红')
    stack.push('绿')
    stack.push('蓝')
    stack.push('黄')
    print(stack.pop())
    print(stack.isEmpty())

